SPHINXsys/src/shared/bodies/body_relation.cpp

```diff
@@ -190,20 +190,19 @@ namespace SPH
 		}
 	}
 	//=================================================================================================//
-	void ReducedInnerBodyRelation::updateConfiguration()
+	void GenerativeBodyRelationInner::updateConfiguration()
 	{
-		tree_->searchNeighborsByParticles(sph_body_->number_of_particles_, *base_particles_,
-			inner_configuration_, get_particle_index_, get_single_search_range_, get_inner_neighbor_);
+		generative_structure_->buildParticleConfiguration(*base_particles_, inner_configuration_);
 	}
 	//=================================================================================================//
-	PartContactBodyRelation::PartContactBodyRelation(BodyPart* body_part, RealBodyVector contact_bodies)
-		: ContactBodyRelation(body_part->getBody(), contact_bodies), body_part_(body_part),
+	BodyPartRelationContact::BodyPartRelationContact(BodyPart* body_part, RealBodyVector contact_bodies)
+		: BodyRelationContact(body_part->getBody(), contact_bodies), body_part_(body_part),
 		body_part_particles_(dynamic_cast<BodyPartByParticle*>(body_part)->body_part_particles_),
 		get_body_part_particle_index_(dynamic_cast<BodyPartByParticle*>(body_part)->body_part_particles_)
 	{
 	}
 	//=================================================================================================//
-	void PartContactBodyRelation::updateConfiguration()
+	void BodyPartRelationContact::updateConfiguration()
 	{
 		size_t number_of_particles = body_part_particles_.size();
 		for (size_t k = 0; k != contact_bodies_.size(); ++k) 
```

Comment:
Actually, if there is arithmetic operation, I use int always, other than size_t, because arithmetic operation may involving negative values. size_t only used for begin to end loop. I will also transfer these loop by using range in the next update.